### ISSUE TITLE
*** No instances to display ***  Error while doing inference

### ISSUE THREAD

**Issue by atinesh** [matterport/Mask_RCNN#2356]
Hello, I have trained the model on custom dataset for 5 epochs, but while doing inference I am getting below error

[![Screenshot-from-2020-09-07-12-35-03.png](https://i.postimg.cc/Jn2nrwRr/Screenshot-from-2020-09-07-12-35-03.png)](https://postimg.cc/gX3dsTwQ)

I tried the following modifications in the code but error is not resolved

1. Lowered `DETECTION_MIN_CONFIDENCE` in `mrcnn/config.py` [Issue: 2132](https://github.com/matterport/Mask_RCNN/issues/2132)

2. Fixed Division by Zero error in `mrcnn/utils.py` with these modifications `from __future__ import division` [Issue: 515](https://github.com/matterport/Mask_RCNN/pull/515), `shift = np.array([0, 0, 1., 1.])` [Issue: 912](https://github.com/matterport/Mask_RCNN/issues/912)

[Training on Custom dataset](https://github.com/arshren/Mask_RCNN/blob/master/Transfer%20Learning%20Mask%20RCNN-Custom%20dataset.ipynb), [Custom dataset](https://github.com/experiencor/kangaroo)

**Comment by suchiz** [comment]
Hi there,
No instance to display isn't an error. it just says the model didn't detect anything. It's probably because 5 epochs is totally nothing for a training ! You should probably train way more !

**Comment by atinesh** [comment]
> Hi there,
> No instance to display isn't an error. it just says the model didn't detect anything. It's probably because 5 epochs is totally nothing for a training ! You should probably train way more !

But the author [here](https://github.com/arshren/Mask_RCNN/blob/master/Transfer%20Learning%20Mask%20RCNN-Custom%20dataset.ipynb) only trained the model for 5 epochs and my system memory (32GB) also getting full until 5 epochs

**Comment by suchiz** [comment]
Indeed. Do you use the exact same dataset ? I know it is tranfer learning, but honestly, 5 epochs with 131 step per epochs sounds so low to me.

Also, the min confidence is overridden in the 2nd block of the colab. So if you modified the one in mrcnn/config.py, it won't apply in your case. Try to lower this one maybe ?

**Comment by atinesh** [comment]
> Indeed. Do you use the exact same dataset ? I know it is tranfer learning, but honestly, 5 epochs with 131 step per epochs sounds so low to me.
> 
> Also, the min confidence is overridden in the 2nd block of the colab. So if you modified the one in mrcnn/config.py, it won't apply in your case. Try to lower this one maybe ?

Yes Same dataset I am using, Ok I will try and check

**Comment by SantaFlang** [comment]
Hello, I have the same problem, and moreover, when I tried to retrain my model, the train time turned to so long that I have to interrupt the program. But it worked perfectly yesterday. THX for reading, hope for reply.
